src/Symfony/Component/DependencyInjection/Context.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Exception\BadMethodCallException;
+use Symfony\Component\DependencyInjection\Exception\ContextElementNotFoundException;
+
+/**
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+final class Context
+{
+    private $elements;
+    private $locked = false;
+
+    public function __construct(array $context = array())
+    {
+        $this->elements = $context;
+    }
+
+    public function get($key)
```

Comment:
I'll add the `has($key) : bool` for now and wait for more opinions :)
Thank you.